src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -354,6 +358,25 @@ public void runWithReplay(Path path, @Nullable Integer replayId, String featured
         });
   }
 
+  @SneakyThrows
+  private Void askWhetherToStartWithOutMap(Throwable throwable) {
+    JavaFxUtil.assertBackgroundThread();
+
+    CountDownLatch userAnswered = new CountDownLatch(1);
+    AtomicReference<Boolean> proceed = new AtomicReference<>(false);
+    List<Action> actions = Arrays.asList(new Action(i18n.get("replay.ignoreMapNotFound"), event -> {
+          proceed.set(true);
+          userAnswered.countDown();
+        }),
+        new Action(i18n.get("replay.abortAfterMapNotFound"), event -> userAnswered.countDown()));
+    notificationService.addNotification(new ImmediateNotification(i18n.get("replay.mapDownloadFailed"), i18n.get("replay.mapDownloadFailed.wannaContinue"), Severity.WARN, actions));
+    userAnswered.await();
+    if (!proceed.get()) {
+      throw throwable;
```

Comment:
I'd throw the exception no matter whether the user proceeds or not.

Reply:
No u don't get the hack I am doing here right... Maybe I should log it that is a good idea but what u suggest is just not working